examples/lightgbm_link_baseline.py

```diff
@@ -54,12 +54,12 @@
 # Prepare col_to_stype dictioanry mapping between column names and stypes
 # for torch_frame Dataset initialization.
 col_to_stype = {}
-src_entity_table_col_to_stype = dataset2inferred_stypes[args.dataset][
-    task.src_entity_table
-]
-dst_entity_table_col_to_stype = dataset2inferred_stypes[args.dataset][
-    task.dst_entity_table
-]
+src_entity_table_col_to_stype = copy.deepcopy(
+    dataset2inferred_stypes[args.dataset][task.src_entity_table]
+)
+dst_entity_table_col_to_stype = copy.deepcopy(
+    dataset2inferred_stypes[args.dataset][task.dst_entity_table]
+)
```

Comment:
Could you clarify why deep copy is required here? `dataset2inferred_stypes[args.dataset]` is never used later, so it seems fine to modify it in-place.

Reply:
Because there are cases where `task.src_entity_table == task.dst_entity_table` which makes `src_entity_table_col_to_stype` and `dst_entity_table_col_to_stype` points to the same object, which will result in error in downstream logics.

Reply:
What is the error? Are we modifying this dict in-place somewhere? That's probably something we wanna do.

Reply:
When src table and dst table are the same table the error will arise. e.g. in rel-stack post-post-related task